docs/entity-cheat-sheet.rst

```diff
@@ -0,0 +1,333 @@
+     - ${create_or_update} = 'create'
+     - ${create_or_update} = 'update'
+     - coalesce(${existing_item}, uuid())
+
+.. note::
+   Current limitation: Only one entity list can be updated per form and each submission can only create or update a single entity.
+
+
+Saving the Entity ID in a Registration Form
+___________________________________________
+
+Depending on your workflow, it may be helpful to save the Entity ID (UUID) in the submission data where the entity is created. 
+
+.. list-table:: Example of saving the ID of a new entity in the submission.
+   :widths: 40 30 30
+   :header-rows: 1
+
+   * - ``type``
+     - ``name``
+     - ``calculation``
+   * - calculate
+     - new_entity_id
+     - ``/data/meta/entity/@id``
+     
+
+Using Entity Data
+-----------------
+
+Entity lists are used just like CSV attachments. You can use multiple entity lists in a single form. There are two main ways to attach an entity list where **listname** is the name of your entity list:
+
+#. Use ``select_one_from_file listname.csv`` or ``select_multiple_from_file listname.csv`` 
+
+   * The **.csv** extension after **listname** is necessary.
+
+#. Use ``csv-external`` with ``listname``
+
+.. note::
+  When you upload your form to Central, it will check the expected attachments and automatically connect an entity list in place of an attachment when the name matches exactly. You can check what entity lists your forms are using by looking at those forms' attachments on Central.
+
+
+Selecting an Entity
+______________________________
+
+When you use ``select_one_from_file listname.csv``, this form field you write in the ``name`` column will hold the ID of your selected entity. This ID is the UUID that Central uses to uniquely track the entity, e.g. ``4d6a1fe1-6dff-4f72-b122-1413fe9b2dd0``. You might notice UUIDs like this in your submission data.
+
+.. list-table:: Example ``survey`` sheet for selecting an entity with ``select_one_from_file``.
+   :widths: 40 30 30
+   :header-rows: 1
+
+   * - ``type``
+     - ``name``
+     - ``label``
+   * - select_one_from_file households.csv
+     - hh_id
+     - Select household
+
+
+Looking up an Entity from an External CSV
+______________________________________
+
+Another way to choose an entity from a list is by another key. Note that the ``calculate`` to get the ``name`` (also referred to as Entity ID or UUID) is only required if you need to update the entity. 
+
+.. list-table:: Example of selecting a household by a barcode ID.
+   :widths: 40 20 10 30
+   :header-rows: 1
+
+   * - ``type``
+     - ``name``
+     - ``label``
+     - ``calculation``
+   * - csv-external
+     - households
+     - 
+     - 
+   * - barcode
+     - barcode
+     - Scan household barcode
+     -
+   * - calculate
+     - hh_id
+     - 
+     - instance("households")/root/item[id=${barcode}]/name
+
+
+
+Updating a Selected Entity
+__________________________
+
+This UUID is the ID that Central needs when updating the entity.
```

Comment:
Which UUID? The one described in the two sections above? Maybe something like "the id from a `select_one_from_file` or the `name` property described in the sections above is the id that..."